dapps/shop/backend/routes/auth.js

```diff
@@ -1,7 +1,170 @@
-const auth = require('./_basicAuth')
+const { Sellers, Shops } = require('../data/db')
+const {
+  createSalt,
+  hashPassword,
+  AuthSeller,
+  authenticated,
+  authenticatedAsSeller
+} = require('./_combinedAuth')
+const { IS_PROD } = require('../utils/const')
+const encConf = require('../utils/encryptedConfig')
+const { validateConfig, validateShop } = require('../utils/validators')
 
 module.exports = function(app) {
-  app.get('/auth', auth, (req, res) => {
+  app.get('/auth', (req, res) => {
+    res.json({ success: req.isAuthenticated() })
+  })
+
+  app.get('/auth/:email', async (req, res) => {
+    // TODO: Add some rate limiting here
+    const seller = await Sellers.findOne({
+      where: {
+        email: req.params.email
+      }
+    })
+    res.json({
```

Comment:
Can we do this with just status codes? E.g

```
const statusCode = seller !== null ? 204 : 404
res.send(statusCode)
```

The same applies to the other requests. It is generally easier to check the status code rather than parse response body to determine success/failure. You can also just return the requested data as the JSON response rather than include information about the request status.

Kind of a stylistic nit but I find it makes client code simpler. I'm happy to do the work to change it since I'm being a pain in the ass. :)

Reply:
Sure, can do this.